support-workers/src/main/scala/com/gu/stripe/StripeService.scala

```diff
@@ -8,23 +8,36 @@ import com.gu.support.config.StripeConfig
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class StripeService(config: StripeConfig, client: FutureHttpClient, baseUrl: String = "https://api.stripe.com/v1")(implicit ec: ExecutionContext)
+class StripeService(val config: StripeConfig, client: FutureHttpClient, baseUrl: String = "https://api.stripe.com/v1")(implicit ec: ExecutionContext)
     extends WebServiceHelper[Stripe.StripeError] {
 
   // Stripe URL is the same in all environments
   val wsUrl = baseUrl
   val httpClient: FutureHttpClient = client
 
-  def createCustomer(token: String, currency: Currency): Future[Customer] =
+  def withCurrency(currency: Currency): StripeServiceForCurrency = new StripeServiceForCurrency(this, currency)
```

Comment:
I had to add this back because we have to pass in the mock StripeServiceForCurrency, but as it's an end to end test so we have to pass it in from the top level.  The simplest way of doing so is to attach it to Services via this method and it all happens by magic.